src/test/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParamsTest.java

```diff
@@ -170,4 +173,21 @@ public void shouldApplyToSettledDateCorrectly() {
         assertThat(transactionSearchParams.getQueryMap().get("to_settled_date"), is(ZonedDateTime.parse("2020-09-27T00:00:00.000Z")));
         assertThat(transactionSearchParams.buildQueryParamString(1L), containsString("to_settled_date=2020-09-26"));
     }
-}
\ No newline at end of file
+
+    @ParameterizedTest
+    @CsvSource({"?, %3F", 
+            "{ , %7B", 
+            "[ , %5B", 
+            "f{o{o}{ , f%7Bo%7Bo%7D%7B", 
+            "foo&, foo%26", 
+            "foo@ , foo%40", 
+            "foo@@ , foo%40%40", 
+            "foo=bar&baz=quux , foo%3Dbar%26baz%3Dquux", 
+            "foo bar , foo+bar"
```

Comment:
Can we also have one where we’re searching for a literal `%` sign (i.e. double-encoding)? And can you test that manually as well please?